File: 2024-2/0684/Unidad 02.a - POO/java/Publicaciones_v04/src/publicaciones_v04/Publicacion.java
package publicaciones_v04;

public abstract class Publicacion {
    public static Integer orden_presentacion = 1;
    protected String nombre;
    protected String autores[];
    protected Integer cantidad_autores;
    protected Integer año;

    public Publicacion(String nombre, String[] autores, Integer cantidad_autores, Integer año) {
        this.nombre = nombre;
        this.autores = autores;
        this.cantidad_autores = cantidad_autores;
        this.año = año;
    }
    
    public Publicacion(String nombre, String autor, Integer año) {
        this.nombre = nombre;
        this.autores = new String[1];
        this.autores[0] = autor;
        this.cantidad_autores = 1;
        this.año = año;
    }

    public abstract void referenciar();    
    
}

File: 2024-2/0682/Unidad 02.a - POO/C#/Publicaciones_v01/Publicaciones_v01/Publicacion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones_v01
{
    public class Publicacion
    {
        private string nombre;
        private string[] autores;
        private int cantidad_autores;
        private int año;

        public Publicacion(string nombre, string[] autores, int cantidad_autores, int año)
        {
            this.nombre = nombre;
            this.autores = autores;
            this.cantidad_autores = cantidad_autores;
            this.año = año;
        }

        public Publicacion(string nombre, string autor, int año)
        {
            this.nombre = nombre;
            this.autores = new string[1];
            this.autores[0] = autor;
            this.cantidad_autores = 1;
            this.año = año;
        }
    }
}
